custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -775,24 +776,70 @@ def _run_inside_docker(self, options, run_mode, raw_arguments):
         return retcode
 
     def _maybe_build_image(self, docker_image_or_directory):
-        ret_docker_image = None
+        def _get_requirements_lines(reqs_file_path):
+            if not os.path.exists(reqs_file_path):
+                return None
+
+            with open(reqs_file_path) as f:
+                lines = f.readlines()
+                lines = [l.strip() for l in lines]
+            return lines
 
+        ret_docker_image = None
         if os.path.isdir(docker_image_or_directory):
             docker_image_or_directory = os.path.abspath(docker_image_or_directory)
+            # Set image tag to the dirname of the docker context.
+            # This may be confusing if try to build different images from different contexts,
+            # with the same folder name:
+            # /path1/my_env
+            # /path2/my_env
+            #
+            # If image with the tag `my_env` exists, it will be untagged.
+            tag = os.path.basename(os.path.abspath(docker_image_or_directory))
+
+            lines = _get_requirements_lines(os.path.join(self.options.code_dir, "requirements.txt"))
+            temp_context_dir = None
+            if lines is not None and not self.options.skip_deps_install:
+                temp_context_dir = tempfile.mkdtemp()
+                shutil.rmtree(temp_context_dir)
+                shutil.copytree(docker_image_or_directory, temp_context_dir)
+                msg = (
+                    "Requirements file has been found in the code dir. DRUM will try to install dependencies into a docker image.\n"
+                    "Docker context has been copied from: {} to: {}".format(
+                        docker_image_or_directory, temp_context_dir
+                    )
+                )
+
+                print(msg)
+                self.logger.debug(msg)
+                docker_image_or_directory = temp_context_dir
+
+                with open(os.path.join(temp_context_dir, "Dockerfile"), mode="a") as f:
+                    if self.options.language == RunLanguage.PYTHON.value:
+                        f.write("\nRUN pip3 install {}".format(" ".join(lines)))
+                    elif self.options.language == RunLanguage.R.value:
+                        quoted_lines = ["'{}'".format(ll) for ll in lines]
+                        deps_str = ", ".join(quoted_lines)
+                        l1 = "\nRUN echo \"r <- getOption('repos'); r['CRAN'] <- 'http://cran.rstudio.com/'; options(repos = r);\" > ~/.Rprofile"
```

Comment:
i don't think it's immediately necessary, but it might be nice to make the cran mirror configurable